BlinkID/lib/microblink_scanner_widget.dart

```diff
@@ -65,10 +78,34 @@ class _MicroblinkScannerWidgetState extends State<MicroblinkScannerWidget> {
     widget.onDetectionStatusUpdate(detectionStatusUpdate.detectionStatus);
   }
 
+  void _onScanDone(MethodCall call) {
+    final name = call.arguments as String;
+    late final RecognizerResultState resultState;
+    if (name == 'UNSUCCESSFUL') {
+      resultState = RecognizerResultState.empty;
+    } else if (name == 'PARTIAL') {
+      resultState = RecognizerResultState.uncertain;
+    } else if (name == 'SUCCESSFUL') {
+      resultState = RecognizerResultState.valid;
+    } else if (name == 'STAGE_SUCCESSFUL') {
+      resultState = RecognizerResultState.stageValid;
+    } else {
+      throw PlatformException(
+        code: 'Unsupported',
+        details: 'Unsupported name: $name',
```

Comment:
Name shouldn't be tied to the android implementation, use the name of the enum instead. (`empty`, `uncertain` etc.).

Please also use a switch statement.